src/billing/Workspaces/Workspaces.tsx

```diff
@@ -222,58 +269,157 @@ interface WorkspacesProps {
+      const endDate = new Date().toISOString().slice(0, 10);
+      const aggregationKeys = ['Workspace~Category'];
+
+      setUpdating(true);
+      Ajax(signal)
+        .Billing.getSpendReport({
+          billingProjectName: billingProject.projectName,
+          startDate,
+          endDate,
+          aggregationKeys,
+        })
+        .then((spend: SpendReportServerResponse) => {
+          const spendDataMap = _.keyBy(
+            (spendItem: WorkspaceSpendData) => `${spendItem.workspace.namespace}-${spendItem.workspace.name}`,
+            (spend.spendDetails[0] as AggregatedWorkspaceSpendData).spendData
+          );
+
+          const updatedWorkspaces = _.map((workspace) => {
+            const key = `${workspace.namespace}-${workspace.name}`;
+            const spendItem = spendDataMap[key];
+
+            if (spendItem) {
+              const costFormatter = new Intl.NumberFormat(navigator.language, {
+                style: 'currency',
+                currency: spendItem?.currency,
+              });
+
+              return {
+                ...workspace,
+                totalSpend: costFormatter.format(parseFloat(spendItem?.cost ?? '0.00')),
+                totalCompute: costFormatter.format(
+                  parseFloat(_.find({ category: 'Compute' }, spendItem?.subAggregation?.spendData)?.cost ?? '0.00')
+                ),
+                totalStorage: costFormatter.format(
+                  parseFloat(_.find({ category: 'Storage' }, spendItem?.subAggregation?.spendData)?.cost ?? '0.00')
+                ),
+              };
+            }
+
+            return workspace;
+          }, workspacesInProject);
+
+          setAllWorkspacesInProject(updatedWorkspaces);
+          setUpdating(false);
+        })
+        .catch(() => setUpdating(false));
+    };
+
+    getSpendReportByWorkspaceAndCategory(selectedDays, workspacesInProject);
+  }, [billingProject.projectName, selectedDays, signal, workspacesInProject]);
+
+  // Apply filters to WorkspacesInProject
+  const searchValueLower = searchValue.toLowerCase();
+  const filteredWorkspacesInProject = _.filter(
+    (workspace: WorkspaceInfo) =>
+      workspace.name.toLowerCase().includes(searchValueLower) ||
+      workspace.googleProject?.toLowerCase().includes(searchValueLower) ||
+      workspace.bucketName?.toLowerCase().includes(searchValueLower),
+    allWorkspacesInProject
+  );
+
+  return (
+    <>
+      {isFeaturePreviewEnabled(SPEND_REPORTING) && (
+        <div
+          style={{
+            display: 'grid',
+            gridTemplateColumns: 'repeat(3, minmax(max-content, 1fr))',
+            rowGap: '1.66rem',
+            columnGap: '1.25rem',
+          }}
+        >
+          <DateRangeFilter
+            label='Date range'
+            rangeOptions={[7, 30, 90]}
+            defaultValue={selectedDays}
+            style={{ gridRowStart: 1, gridColumnStart: 1 }}
+            onChange={setSelectedDays}
+          />
```

Comment:
Can we refactor the existing report to use this filter as well so the two reports are using the same component?

Reply:
Done: https://github.com/DataBiosphere/terra-ui/pull/5165/commits/24aae6b17f45632e6fe9bcf108697679d735a3f5